File: RentaTeamTask/app/src/main/java/com/example/rentateamtask/view/MainActivity.java
package com.example.rentateamtask.view;

import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import com.example.rentateamtask.R;
import com.example.rentateamtask.view.fragments.AboutFragment;
import com.example.rentateamtask.view.fragments.UserListFragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {
    Fragment userListFragment = new UserListFragment();
    Fragment aboutFragment = new AboutFragment();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        BottomNavigationView bottomNavigationView = findViewById(R.id.bottomNavigationView);
        bottomNavigationView.setOnNavigationItemSelectedListener(navListener);
    }

    public BottomNavigationView.OnNavigationItemSelectedListener navListener = item -> {
        Fragment selectedFragment;
        if (item.getItemId() == R.id.page1)
            selectedFragment = userListFragment;
        else
            selectedFragment = aboutFragment;

        getSupportFragmentManager()
                .beginTransaction()
                .setCustomAnimations(
                        R.anim.fade_in,
                        R.anim.fade_out)
                .replace(R.id.fragmentContainerView, selectedFragment)
                .commit();
        return true;
    };

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
}

File: CurrencyExchanger/app/src/main/java/com/example/currencyexchanger/presentation/views/MainActivity.kt
package com.example.currencyexchanger.presentation.views

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.example.currencyexchanger.R
import com.example.currencyexchanger.data.RepositoryImpl
import com.example.currencyexchanger.data.converter.EntityConverter
import com.example.currencyexchanger.data.network.NetworkModule
import com.example.currencyexchanger.databinding.ActivityMainBinding
import com.example.currencyexchanger.di.AppResLocator
import com.example.currencyexchanger.domain.interactor.InteractorImpl
import com.example.currencyexchanger.presentation.fragments.FavouritesFragment
import com.example.currencyexchanger.presentation.fragments.PopularFragment
import com.example.currencyexchanger.presentation.viewmodel.CurrencyViewModel
import com.google.android.material.bottomnavigation.BottomNavigationView

/**
 * @author Bulat Bagaviev
 * @created 12.10.2022
 */
class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private lateinit var mainViewModel: CurrencyViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        initViews()
    }

    override fun onStart() {
        createViewModel()
        super.onStart()
    }

    private fun initViews() {
        with(binding) {
            bottomNavigationView.setOnItemSelectedListener(createListener())
        }
    }

    private fun createListener() =
        BottomNavigationView.OnNavigationItemSelectedListener { item ->
            val fragment: Fragment = if (item.itemId == R.id.page1)
                PopularFragment.newInstance()
            else
                FavouritesFragment.newInstance()

            supportFragmentManager
                .beginTransaction()
                .setCustomAnimations(R.anim.fade_in, R.anim.fade_out)
                .replace(R.id.fragmentContainerView, fragment)
                .commit()

            true
        }

    private fun createViewModel() {
        val interactor = InteractorImpl(RepositoryImpl(NetworkModule(), EntityConverter()))

        mainViewModel = ViewModelProvider(this, object: ViewModelProvider.Factory {
            override fun <T : ViewModel> create(modelClass: Class<T>): T {
                return CurrencyViewModel(interactor, (applicationContext as AppResLocator)) as T
            }
        }).get(CurrencyViewModel::class.java)
    }
}